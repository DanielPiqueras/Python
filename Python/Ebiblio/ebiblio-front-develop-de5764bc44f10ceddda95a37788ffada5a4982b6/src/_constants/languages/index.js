export { default as languages } from './languages';
export {
  DEFAULT_LANGUAGE,
  getAllCodes,
  getAllLocales,
  languageFindByCode,
  languageFindByLocale,
  languageFindByName,
} from './utils';
