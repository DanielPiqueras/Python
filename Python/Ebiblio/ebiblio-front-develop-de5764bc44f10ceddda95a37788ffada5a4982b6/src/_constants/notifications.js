export const types = Object.freeze({
  SUCCESS: 'success',
  ERROR: 'error',
  INFO: 'info',
});
