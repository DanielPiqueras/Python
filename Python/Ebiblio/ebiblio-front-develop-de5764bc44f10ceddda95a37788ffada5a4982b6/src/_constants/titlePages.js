export default [
  {
    key: '/admin/users/create',
    title: 'pages:adminUserNew',
  },
  {
    key: '/admin/books/create',
    title: 'pages:adminBookNew',
  },
  {
    key: '/admin/(users|books)/history/.+',
    title: 'pages:adminHistoryOf',
  },
  {
    key: '/admin/books',
    title: 'pages:adminBookList',
  },
  {
    key: '/admin/users/.+',
    title: 'pages:adminUserUpdate',
  },
  {
    key: '/admin/users',
    title: 'pages:adminUserList',
  },
  {
    key: '/admin/history',
    title: 'pages:adminBookHistory',
  },
  {
    key: '/profile',
    title: 'pages:profile',
  },
  {
    key: '/history',
    title: 'pages:personalHistory',
  },
  {
    key: '/books',
    title: 'pages:bookList',
  },
  {
    key: '/logout',
    title: 'pages:logout',
  },
  {
    key: '/',
    title: 'pages:login',
  },
];
