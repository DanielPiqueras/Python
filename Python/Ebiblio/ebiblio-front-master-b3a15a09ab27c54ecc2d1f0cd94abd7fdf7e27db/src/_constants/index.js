export { bookConstants } from './book.constants';
export { historyBookConstants } from './historyBook.constants';
export {
  DEFAULT_LANGUAGE,
  getAllCodes,
  getAllLocales,
  languageFindByCode,
  languageFindByLocale,
  languageFindByName,
  languages,
} from './languages';
export { admin as adminNavLinks, sections, user as userNavLinks } from './navlinks';
export { types as notificationsTypes } from './notifications';
export { default as titlePages } from './titlePages';
export { userConstants } from './user.constants';
export { APP_NAME } from './app';
