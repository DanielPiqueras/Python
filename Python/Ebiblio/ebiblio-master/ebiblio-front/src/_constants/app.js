export const APP_NAME = 'Sewan Biblio';
