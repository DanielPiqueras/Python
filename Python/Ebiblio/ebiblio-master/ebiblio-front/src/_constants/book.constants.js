export const bookConstants = Object.freeze({
  GETINFO_REQUEST: 'BOOKS_GETINFO_REQUEST',
  GETINFO_SUCCESS: 'BOOKS_GETINFO_SUCCESS',
  GETINFO_FAILURE: 'BOOKS_GETINFO_FAILURE',

  GETALL_REQUEST: 'BOOKS_GETALL_REQUEST',
  GETALL_SUCCESS: 'BOOKS_GETALL_SUCCESS',
  GETALL_FAILURE: 'BOOKS_GETALL_FAILURE',

  CREATE_REQUEST: 'BOOKS_CREATE_REQUEST',
  CREATE_SUCCESS: 'BOOKS_CREATE_SUCCESS',
  CREATE_FAILURE: 'BOOKS_CREATE_FAILURE',

  UPDATE_REQUEST: 'BOOKS_UPDATE_REQUEST',
  UPDATE_SUCCESS: 'BOOKS_UPDATE_SUCCESS',
  UPDATE_FAILURE: 'BOOKS_UPDATE_FAILURE',

  DELETE_REQUEST: 'BOOKS_DELETE_REQUEST',
  DELETE_SUCCESS: 'BOOKS_DELETE_SUCCESS',
  DELETE_FAILURE: 'BOOKS_DELETE_FAILURE',

  CLEAR_INFO: 'CLEAR_INFO',
});
