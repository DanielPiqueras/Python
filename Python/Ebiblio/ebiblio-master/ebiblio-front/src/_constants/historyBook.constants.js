export const historyBookConstants = Object.freeze({
  GETALL_REQUEST: 'HISTORY_BOOKS_GETALL_REQUEST',
  GETALL_SUCCESS: 'HISTORY_BOOKS_GETALL_SUCCESS',
  GETALL_FAILURE: 'HISTORY_BOOKS_GETALL_FAILURE',

  GETALL_USER_REQUEST: 'HISTORY_BOOKS_GETALL_USER_REQUEST',
  GETALL_USER_SUCCESS: 'HISTORY_BOOKS_GETALL_USER_SUCCESS',
  GETALL_USER_FAILURE: 'HISTORY_BOOKS_GETALL_USER_FAILURE',

  CREATE_REQUEST: 'HISTORY_BOOKS_CREATE_REQUEST',
  CREATE_SUCCESS: 'HISTORY_BOOKS_CREATE_SUCCESS',
  CREATE_FAILURE: 'HISTORY_BOOKS_CREATE_FAILURE',
});
