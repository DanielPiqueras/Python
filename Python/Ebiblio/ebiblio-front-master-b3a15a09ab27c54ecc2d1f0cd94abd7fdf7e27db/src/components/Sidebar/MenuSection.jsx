/** Packages */
import PropTypes from 'prop-types';
import { groupBy, prop } from 'ramda';
import React from 'react';
import { useTranslation } from 'react-i18next';
import styled from 'styled-components';
import Icon from '@material-ui/core/Icon';

/** Modules */
import CustomNavLink from './CustomNavLink';
import { sections } from '../../_constants';

const StyledHeader = styled.p`
  text-decoration: none;
  display: flex;
  justify-content: center;
  padding: 4px 8px;
  color: gray;
  cursor: default;
  transition: all 0.25s ease-out;
`;

const MenuSection = ({ isOpen, subsections }) => {
  const { t } = useTranslation();
  const filteredSubsections = groupBy(prop('section'), subsections);
  const Sections = Object.keys(filteredSubsections).reduce((acc = [], key) => {
    const section = sections[key];
    const translatedTitle = t(section.text);
    const currentSection = filteredSubsections[key];
    const links = Object.values(currentSection)
      .map(current => <CustomNavLink key={current.to} isOpen={isOpen} {...current} />);
    acc.push(
      <div key={key}>
        <StyledHeader fontSize="default">
          {isOpen
            ? translatedTitle
            : section.icon && (
            <Icon title={translatedTitle} alt={translatedTitle} fontWeight="bold">
              {section.icon}
            </Icon>
            )}
        </StyledHeader>
        {links}
      </div>,
    );
    return acc;
  }, []);
  return Sections;
};

MenuSection.displayName = 'MenuSection';
MenuSection.propTypes = {
  subsections: PropTypes.arrayOf(
    PropTypes.shape({
      section: PropTypes.string.isRequired,
      to: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
      icon: PropTypes.object.isRequired,
    }),
  ),
};

/** Export */
export default React.memo(MenuSection);
