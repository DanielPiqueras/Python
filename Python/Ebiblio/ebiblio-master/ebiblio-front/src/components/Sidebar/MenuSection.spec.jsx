import { mount } from 'enzyme';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';

import { adminNavLinks } from '../../_constants';
import MenuSection from './MenuSection';

describe('MenuSection', () => {
  test('Component should be mount correctly', () => {
    const wrapper = mount(
      <MemoryRouter>
        <MenuSection subsections={adminNavLinks} />
      </MemoryRouter>,
    );
    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });
});
