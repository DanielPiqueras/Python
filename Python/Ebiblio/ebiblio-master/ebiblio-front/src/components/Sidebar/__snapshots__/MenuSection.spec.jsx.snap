// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuSection Component should be mount correctly 1`] = `ReactWrapper {}`;
